core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java

```diff
@@ -13,39 +13,30 @@
 package org.polarsys.capella.core.ui.semantic.browser.label.provider;
 
 import org.eclipse.jface.viewers.ILabelProvider;
-import org.eclipse.jface.viewers.StyledString;
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.ui.internal.navigator.NavigatorDecoratingLabelProvider;
-import org.polarsys.capella.common.ui.toolkit.browser.content.provider.wrapper.BrowserElementWrapper;
+import org.polarsys.capella.common.ui.toolkit.browser.content.provider.wrapper.CategoryWrapper;
 import org.polarsys.capella.common.ui.toolkit.browser.content.provider.wrapper.EObjectWrapper;
+import org.polarsys.capella.common.ui.toolkit.browser.content.provider.wrapper.PrimitiveWrapper;
 
-public class SemanticBrowserDecoratingLabelProvider extends  NavigatorDecoratingLabelProvider{
+@SuppressWarnings("restriction")
+public class SemanticBrowserDecoratingLabelProvider extends NavigatorDecoratingLabelProvider {
 
   public SemanticBrowserDecoratingLabelProvider(ILabelProvider commonLabelProvider) {
     super(commonLabelProvider);
   }
-  @Override
-  protected StyledString getStyledText(Object element) {
-    if (element instanceof EObjectWrapper) {
-      element = ((BrowserElementWrapper) element).getElement();
-    }
-    return super.getStyledText(element);
-  }
 
   @Override
   public Image getImage(Object element) {
-    if (element instanceof EObjectWrapper) {
-      element = ((BrowserElementWrapper) element).getElement();
-    }
-    return super.getImage(element);
-  }
+    // Do not unwrap category or primitive wrappers, as they have a specific image
+    if (element instanceof CategoryWrapper || element instanceof PrimitiveWrapper) {
```

Comment:
a bit weird to have a empty if.
maybe move the comment below, or a !(element CategoryWapper && !xxx)

Reply:
removed it totally as it was irrelevant